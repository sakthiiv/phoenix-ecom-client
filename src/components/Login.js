import React from "react";
import { Redirect } from "react-router-dom";
class Login extends React.Component {
    state = {
      redirectToReferrer: false
    };
  
    // login = () => {
    //   fakeAuth.authenticate(() => {
    //     this.setState({ redirectToReferrer: true });
    //   });
    // };
    handleSubmit = (e) => {
        e.preventDefault();
        this.props.statusLogin(true);
    }
    render() {
    //   const { from } = this.props.location.state || { from: { pathname: "/" } };
    //   const { redirectToReferrer } = this.state;
  
    //   if (redirectToReferrer) {
    //     return <Redirect to={from} />;
    //   }
      const {showModal} = this.props;
      return (
        <div className={showModal ? '':'hidden'}>
          <p>You must log in</p>
          <form onSubmit={this.handleSubmit}>
                <div className="form-group">
                    <label htmlFor="cname">Enter username</label>
                    <input id='uname' name="uname" type="text" placeholder='Enter username' className="form-control"  required></input>
                </div>
                <div className="form-group">
                    <label htmlFor="cname">Enter password</label>
                    <input id='password' name="password" type="password" placeholder='Enter password' className="form-control"  required></input>
                </div>
                <input type="submit" value="Submit" className="btn-success" /> 
           </form>
        </div>
      );
    }
  }

  export default Login;