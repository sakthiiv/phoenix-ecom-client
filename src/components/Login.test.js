import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';
describe('Main', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<Login />);
      expect(wrapper).toMatchSnapshot();
      // On the first run of this test, Jest will generate a snapshot file automatically.
    });
});