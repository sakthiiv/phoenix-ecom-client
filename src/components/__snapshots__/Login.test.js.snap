// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="hidden"
>
  <p>
    You must log in
  </p>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="cname"
      >
        Enter username
      </label>
      <input
        className="form-control"
        id="uname"
        name="uname"
        placeholder="Enter username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="cname"
      >
        Enter password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Enter password"
        required={true}
        type="password"
      />
    </div>
    <input
      className="btn-success"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
